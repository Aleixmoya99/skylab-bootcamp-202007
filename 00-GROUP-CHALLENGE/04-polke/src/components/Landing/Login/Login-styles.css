.Login {
	border: 1px solid black;
	height: 60vh;
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
}

.Login button {
	width: 75vw;
}

@media only screen and (min-width: 500px) and (max-width: 682px) {
	.Login {
	}
	.Login__form {
		width: 90vw;
		border: 1px solid red;
		display: flex;
		flex-direction: column;
		justify-content: center;
		align-items: center;
	}
	.Login button {
		width: 55vw;
	}
	.form-control {
		width: 74%;
	}
}

@media only screen and (min-width: 682px) and (max-width: 1600px) {
	.Login {
		width: 90vw;
		border: 1px solid red;
		display: flex;
		flex-direction: column;
		justify-content: center;
		align-items: flex-start;
	}
	.Login__form {
		width: 50vw;
		border: 1px solid red;
		display: flex;
		flex-direction: column;
		justify-content: center;
		align-items: center;
	}

	.Login button {
		width: 150px;
	}
	.form-control {
		width: 74%;
	}
}
