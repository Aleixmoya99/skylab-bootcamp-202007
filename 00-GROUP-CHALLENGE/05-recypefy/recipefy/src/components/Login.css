.main__boxes{
    display: none;
}

.login__container{
    width: 100vw;
    margin-top: 80px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
 
}  

.login__box{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background-color: #efee9d70;
    width: fit-content;
    padding: 16px;
    border-radius: 10px;
}

form{
    text-align: center;
}

.form__input{
    margin-top: 5px;
    margin-bottom: 15px;
    height: 30px;
    min-width: 250px;
    width: 20vw;
    padding: 2px;
    border-radius: 4px;
    border: none;
    background-color: #dddb57;
    text-align: center;
    font-size: 18px;
}

.login__button{
    background: none;
	background-color: #efee9d;
	border: 1px solid black;
	padding: 8px 16px;
	font: inherit;
	cursor: pointer;
    outline: inherit;
    border-radius: 4px;
}

.login__container p {
    border-top: 1px solid black;
    border-bottom: 1px solid black;
    margin: 10px;
    width: 10vw;
    text-align: center;
}

.login__container h2{
    margin: 8px;
}

.login__container a{
    padding: 12px;
    margin: 5px;
    background-color: rgba(156, 149, 149, 0.24);
    border-radius: 4px;
    min-width: 80px;
    width: 12vw;
    text-align: center;
    cursor: pointer;
}



footer{
    position: absolute;
    bottom: 0;
}