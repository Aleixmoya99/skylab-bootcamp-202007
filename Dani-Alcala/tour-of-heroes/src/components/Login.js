import React, { useState, useEffect } from 'react'
import { login, logout } from '../actions/authActions'
import authStore from '../stores/authStore'


function Login() {
    const email = 'daniamcode@gmail.com'
    const password = '123456'

    const[isLogged, setIsLogged] = useState(authStore.isLogged())
    const [user, setUser] = useState(authStore.getUserProfile())

    useEffect(() => {
        authStore.addChangeListener(onAuthChange)
        
        return () => authStore.removeChangeListener(onAuthChange)
    }, [isLogged, user])

    function onAuthChange() {
        setIsLogged(authStore.isLogged())
        setUser(authStore.getUserProfile())
    }

    return (
		<>
        {!isLogged && (
			<button onClick={() => login(email, password)}>Login</button>
        )}
        {isLogged && 
            <button onClick={() =>logout()}>Logout</button>}
            <img src={user && user.photoURL} />
            <p>{user && user.email}</p>
		</>
	);
}

export default Login