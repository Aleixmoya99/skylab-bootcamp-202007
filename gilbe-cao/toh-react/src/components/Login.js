import React, { useState, useEffect } from 'react';
import authStore from '../stores/authStore';
import { login, logout } from '../actions/authAtions';

function Login() {
	const [isLogged, setIsLogged] = useState(authStore.getIsLogged());

	useEffect(() => {
		authStore.addChangeListener(onLoginChange);
		return () => authStore.removeChangeListener(onLoginChange);
	}, [isLogged]);

	function onLoginChange() {
		setIsLogged(authStore.getIsLogged());
	}

	return (
		<>
			{isLogged && <button onClick={() => logout()}>Logout</button>}
			{!isLogged && <button onClick={() => login()}>Login</button>}
		</>
	);
}

export default Login;
