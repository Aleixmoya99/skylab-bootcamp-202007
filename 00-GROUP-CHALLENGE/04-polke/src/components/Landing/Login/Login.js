import React from 'react';
import { Form, FormControl, FormGroup, Button, Control } from 'react-bootstrap';
import '../../../shared/generalStyles.css';
import './Login-styles.css';

function Login() {
	return (
		<div className="Login">
			<form className="Login__form">
				<h2>Login</h2>
				<FormGroup controlId="username" bsSize="large">
					<Button>Google</Button>
				</FormGroup>
				<FormGroup controlId="password" bsSize="large">
					<Button>GitHub</Button>
				</FormGroup>
				<h4>User</h4>
				<FormControl />
				<h4>Pasword</h4>
				<FormControl />
				<FormGroup className="" controlId="login" bsSize="large">
					<Button>Enter</Button>
				</FormGroup>
				<a>Register</a>
			</form>
		</div>
	);
}

export default Login;
