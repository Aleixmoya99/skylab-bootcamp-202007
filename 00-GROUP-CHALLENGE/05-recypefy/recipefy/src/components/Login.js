import React, { useState } from 'react';
import './Login.css';

function Login(){

    const[email, setEmail] = useState('')
    const [password, setPassword] = useState('');

    function validateForm(){
        return email.length > 0 && password.length > 0
    }

    

    return(
        <div className="login__container">
            <div className="login__box">
                <form>
                    <label for="email">Email</label><br/>
                    <input autoFocus type="email" name="email" className="form__input" value={email} onChange={e => setEmail(e.target.value)}/><br/>
                    <label>Password</label><br/>
                    <input type="password" className="form__input" value={password} onChange={e => setPassword(e.target.value)}/>
                </form>
                <button type="submit" className="login__button" disabled={!validateForm()}>Login</button>
                <p>or</p>
                <h2>REGISTER</h2>
                <a>GOOGLE</a>
                <a>EMAIL</a>
                <a>GUEST</a>
            </div>
        </div>

    );
}

export default Login;