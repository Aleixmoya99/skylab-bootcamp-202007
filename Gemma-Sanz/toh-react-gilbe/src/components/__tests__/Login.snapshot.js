import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import Login from '../Login';

describe('Login snapshot', () => {
    const loginTree = renderer.create(
        <Router>
            <Login />
        </Router>
    );

    it('should match', () => {
        expect(loginTree.toJSON()).toMatchSnapshot();
    });
});