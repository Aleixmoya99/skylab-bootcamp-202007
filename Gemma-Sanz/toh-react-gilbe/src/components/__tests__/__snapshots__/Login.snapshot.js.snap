// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot should match 1`] = `
<button
  onClick={[Function]}
>
  Login
</button>
`;
